src/main/java/org/hl7/davinci/priorauth/Database.java

```diff
@@ -173,39 +185,55 @@ public IBaseResource read(String resourceType, String id, String patient, String
   }
 
   /**
-   * Insert a resource into the database.
+   * Insert a resource into database.
    * 
-   * @param resourceType - the resource type.
-   * @param id           - the resource id.
+   * @param resourceType - the tpye of the resource.
+   * @param keys         - string array of key (table column) names.
+   * @param values       - string array of values to add into the columns.
    * @param resource     - the resource itself.
    * @return boolean - whether or not the resource was written.
    */
-  public boolean write(String resourceType, String id, String patient, IBaseResource resource) {
+  public boolean write(String resourceType, String[] keys, String[] values, IBaseResource resource) {
```

Comment:
Looking for feedback on this - having an array of keys and an array of values instead of a dictionary. This made generating the SQL simpler but I'm curious if you think a dictionary would be better

Reply:
This works, but I would like to see a stronger grouping between the key and value so it is clearer what key goes with which value. This interface could make it easier to put the values into the array in the wrong order and causing the values to go into the wrong columns.

Reply:
Updated to be a dictionary: https://github.com/HL7-DaVinci/prior-auth/commit/4d3ae1bda03adbaba03855e7e8419e8d90ca8785